airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py

```diff
@@ -208,9 +285,14 @@ def _send(self, request: requests.PreparedRequest, request_kwargs: Mapping[str,
 
         # TODO: Consider dynamic retry count depending on subsequent error codes
         elif error_resolution.response_action == ResponseAction.RETRY:
-            custom_backoff_time = self._backoff_strategy.backoff_time(
-                response_or_exception=response if response is not None else exc, attempt_count=self._request_attempt_count[request]
-            )
+            custom_backoff_time = None
+            for backoff_strategy in self._backoff_strategies:
```

Comment:
This loop could make it so that `custom_backoff_time` is none. What happens in that case?

Reply:
If `custom_backoff_time` is None then the `DefaultBackoffException` is raised instead of `UserDefinedBackoffException`.

https://github.com/airbytehq/airbyte/blob/0b04c48f4660aed6e13cb11d152763ea880f783b/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py#L300

Reply:
You are right. Should this be named `user_defined_backoff_time` then to align with the other concepts?

Reply:
Yeah, I agree that makes sense.